server/sockets/list.js

```diff
@@ -477,6 +477,101 @@ const changeListType = (socket, dashboardClients, cohortClients, listClients) =>
       });
   });
 
+const emitListsOnRemoveCohortMember = (socket, dashboardClients, listClients) =>
+  socket.on(CohortActionTypes.REMOVE_MEMBER_SUCCESS, data => {
+    const { cohortId, userId } = data;
+
+    List.find({ cohortId })
+      .populate('cohortId', 'memberIds')
+      .lean()
+      .exec()
+      .then(docs => {
+        if (docs) {
+          const {
+            cohortId: { memberIds: cohortMemberIds }
+          } = docs[0];
+          const listIdsUserWasRemovedFrom = [];
+          const listIdsUserRemained = [];
+
+          docs.forEach(doc => {
+            const { type, viewersIds } = doc;
+            const isListViewer = checkIfArrayContainsUserId(viewersIds, userId);
+            const listId = doc._id.toString();
+
+            if (isListViewer) {
+              listIdsUserRemained.push(listId);
+            } else if (!isListViewer && type === ListType.SHARED) {
+              listIdsUserWasRemovedFrom.push(listId);
+            }
+          });
+
+          if (listIdsUserWasRemovedFrom.length > 0) {
+            if (dashboardClients.has(userId)) {
+              const { socketId } = dashboardClients.get(userId);
+
+              socket.broadcast
+                .to(socketId)
+                .emit(ListActionTypes.REMOVE_BY_IDS, listIdsUserWasRemovedFrom);
```

Comment:
this was unclear for me from the very beginning. Here I just figure it out that this action is responsible for removing the user from lists where he was, so it should be named something like ListActionTypes.REMOVE_USER_FROM_LISTS

Reply:
This action removes lists. If someone removes you from the cohort and you are on the dashboard you have to lose this cohort's list. This event sends an array of lists' ids which have to be removed from the dashboard, we don't use 'LIST' word in List action types names so I think this action's type can be named as `ListActionTypes.REMOVE_BY_IDS`